src/adaptive_threshold.jl

```diff
@@ -77,35 +77,39 @@ See also [`binarize!`](@ref) for in-place operation.
 
 """
 struct AdaptiveThreshold <: AbstractImageBinarizationAlgorithm
+    window_size::Int
     percentage::Float64
 
-    function AdaptiveThreshold(percentage)
+    function AdaptiveThreshold(window_size::Integer, percentage::Real)
+        window_size < 0 && throw(ArgumentError("window_size should be non-negative."))
         (percentage < 0 || percentage > 100) && throw(ArgumentError("percentage should be ∈ [0, 100]."))
-        new(percentage)
+        new(window_size, percentage)
     end
 end
 
-function AdaptiveThreshold(; percentage::Real = 15, window_size=nothing)
-    if window_size === nothing
-        return AdaptiveThreshold(percentage)
-    else
-        # deprecate window_size
-        return AdaptiveThreshold(window_size, percentage)
+function AdaptiveThreshold(img::Union{AbstractArray, Nothing}=nothing;
+                           window_size::Union{Integer, Nothing} = nothing,
+                           percentage::Real = 15)
+    if window_size == nothing
```

Comment:
Currently, it would raise a depwarn when we call `AdaptiveThreshold()`, the idea is to ask users to provide either an image for parameter inference or explicitly pass the parameter. How do you think of this change? @zygmuntszpak